### PR TITLE
Adding reviewscoukcraft plugin

### PR DESCRIPTION
**Pull request by adteulade**
Adding and use reviewscoukcraft plugin in order to get reviews from reviews.co.uk API, modifying the JSON-LD in order to display Homecharge products as Services (Installation of the solo charger and the solo charger) in order that the merchant reviews seems relevant for Google, adding informations about the Service (area served, audience, provider etc...). With the use of Craft CMS `custom fields` feature, allows admin users to specify more information  about the different Home Solo Charger products.

### CODE REVIEW

File: services/SeomaticService.php [hunk 1]
```diff
@@ -2478,6 +2456,143 @@ public function getMainEntityOfPageJSONLD($meta, $identity, $locale, $isMainEnti
...
+            if ($reviewsAverageRating && $reviewsTotalNumber && $reviewsBestRating && $reviewsWorstRating) {
+                $mainEntityOfPageJSONLD['aggregateRating'] = [
+                    'type' => 'AggregateRating',
+                    'ratingValue' => (string)round($reviewsAverageRating, 1),
+                    'reviewCount' => (string)number_format($reviewsTotalNumber),
+                    'bestRating' => (string)$reviewsBestRating,
+                    'worstRating' => (string)$reviewsWorstRating,
+                ];
+            }
+        }
```
Comment (mikefrancis) [inline services/SeomaticService.php:2500]: Blank line after here

File: services/SeomaticService.php [hunk 2]
```diff
@@ -2478,6 +2456,143 @@ public function getMainEntityOfPageJSONLD($meta, $identity, $locale, $isMainEnti
...
+     *
+     * @param \Craft\BaseElementModel $element
+     * @param array                   $identity
+     * @return array
+     */
+    protected function getOffersArrayForJsonLD(BaseElementModel $element, array $identity)
+    {
+        $offers = [];
+
+        if (isset($element->productOffers) && isset($element->productCurrency))
```
Comment (mikefrancis) [inline services/SeomaticService.php]: Braces should be on the same line as the loop/conditional

**Review by mikefrancis — COMMENTED** [review]
I think it is their code but just some formatting

File: services/SeomaticService.php [hunk 3]
```diff
@@ -2432,9 +2432,6 @@ public function getMainEntityOfPageJSONLD($meta, $identity, $locale, $isMainEnti
                     if (isset($settings->reviewsPluginHandle) && isset(craft()->{$settings->reviewsPluginHandle})) {
                         $mainEntityOfPageJSONLD = $this->getAggregrateRatingForJsonLD(craft()->{$settings->reviewsPluginHandle}, $mainEntityOfPageJSONLD, $element, $settings);
                     }
-
-
-
```
Comment (LeeBrooks3) [inline services/SeomaticService.php]: I'd have probably left one of these gaps in

File: services/SeomaticService.php [hunk 4]
```diff
@@ -2568,12 +2566,11 @@ protected function getOffersArrayForJsonLD(BaseElementModel $element, array $ide
...
-        {
-            foreach($element->productOffers as $offer)
-            {
-                if (isset($offer->offerName) && isset($offer->offerPrice))
-                {
+        if (isset($element->productOffers) && isset($element->productCurrency)) {
+
+            foreach($element->productOffers as $offer) {
+
+                if (isset($offer->offerName) && isset($offer->offerPrice)) {
```
Comment (LeeBrooks3) [inline services/SeomaticService.php]: more scope for early returns here...
```php
if (!isset($element->productOffers) || !isset($element->productCurrency)) {
    return $offers;
}

foreach($element->productOffers as $offer) {
        if (!isset($offer->offerName) || !isset($offer->offerPrice)) {
            continue;
        }

        ...
```

Comment (adteulade) [reply services/SeomaticService.php]: I was told that it's better to have one return in a function during my studies, that's why I've got some trouble to get it done as you told me ;-)

**Pull request merged**